Requires curl 8.2.0:
```
checking for explicit_bzero... yes
checking for explicit_memset... no
make[2]: Leaving directory '/build/curl-8.1.2/lib'
make[1]: *** [Makefile:1507: all] Error 2
make[1]: Leaving directory '/build/curl-8.1.2/lib'
make: *** [Makefile:1267: all-recursive] Error 1
error: builder for '/nix/store/msj8l0hwqzav248nypk32q0ilj8b7hb1-curl-8.1.2.drv' failed with exit code 2;
       last 10 log lines:
       > vquic/curl_ngtcp2.c:36:10: fatal error: ngtcp2/ngtcp2_crypto_openssl.h: No such file or directory
       >    36 | #include <ngtcp2/ngtcp2_crypto_openssl.h>
       >       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > compilation terminated.
       > make[2]: *** [Makefile:3274: vquic/libcurl_la-curl_ngtcp2.lo] Error 1
       > make[2]: *** Waiting for unfinished jobs....
       > make[2]: Leaving directory '/build/curl-8.1.2/lib'
       > make[1]: *** [Makefile:1507: all] Error 2
       > make[1]: Leaving directory '/build/curl-8.1.2/lib'
       > make: *** [Makefile:1267: all-recursive] Error 1
       For full logs, run 'nix log /nix/store/msj8l0hwqzav248nypk32q0ilj8b7hb1-curl-8.1.2.drv'.

```

--- a/pkgs/development/libraries/ngtcp2/default.nix
+++ b/pkgs/development/libraries/ngtcp2/default.nix
@@ -8,13 +8,13 @@
 
 stdenv.mkDerivation rec {
   pname = "ngtcp2";
-  version = "0.15.0";
+  version = "0.17.0";
 
   src = fetchFromGitHub {
     owner = "ngtcp2";
     repo = pname;
     rev = "v${version}";
-    hash = "sha256-FWNWpRuCUyqTIyLZkBFKrd2urjSCqHp20mBAXOcJm14=";
+    hash = "sha256-vY3RooC8ttezru6vAqbG1MU5uZhD8fLnlEYVYS3pFRk=";
   };
 
   outputs = [ "out" "dev" "doc" ];
@@ -26,6 +26,13 @@ stdenv.mkDerivation rec {
   cmakeFlags = [
     "-DENABLE_STATIC_LIB=OFF"
   ];
+
+  preConfigure = ''
+    # https://github.com/ngtcp2/ngtcp2/issues/858
+    # Fix ngtcp2_crypto_openssl remnants.
+    substituteInPlace crypto/includes/CMakeLists.txt \
+      --replace 'ngtcp2/ngtcp2_crypto_openssl.h' 'ngtcp2/ngtcp2_crypto_quictls.h'
+  '';
 
   doCheck = true;
   enableParallelBuilding = true;

